clients/drcachesim/tracer/raw2trace_directory.cpp

```diff
@@ -108,17 +109,35 @@ raw2trace_directory_t::open_thread_log_file(const char *basename)
         strcmp(basename, DRMEMTRACE_FUNCTION_LIST_FILENAME) == 0)
         return "";
     // Skip any non-.raw in case someone put some other file in there.
-    const char *basename_pre_suffix = strrchr(basename, '.');
-    if (basename_pre_suffix != nullptr)
-        basename_pre_suffix = strstr(basename_pre_suffix, OUTFILE_SUFFIX);
+    const char *basename_dot = strrchr(basename, '.');
+    if (basename_dot == nullptr)
+        return "";
+    const char *basename_pre_suffix = nullptr;
+#ifdef HAS_ZLIB
+    bool is_gzipped = false;
+    basename_pre_suffix = strstr(basename_dot, OUTFILE_SUFFIX_GZ);
+    if (basename_pre_suffix != nullptr) {
+        is_gzipped = true;
+        basename_dot = strrchr(basename_pre_suffix, '.');
+    }
+#endif
+    if (basename_pre_suffix == nullptr)
+        basename_pre_suffix = strstr(basename_dot, OUTFILE_SUFFIX);
     if (basename_pre_suffix == nullptr)
         return "";
     if (dr_snprintf(path, BUFFER_SIZE_ELEMENTS(path), "%s%s%s", indir_.c_str(), DIRSEP,
                     basename) <= 0) {
         return "Failed to get full path of file " + std::string(basename);
     }
     NULL_TERMINATE_BUFFER(path);
-    in_files_.push_back(new std::ifstream(path, std::ifstream::binary));
+    std::istream *ifile;
+#ifdef HAS_ZLIB
+    if (is_gzipped)
+        ifile = new gzip_istream_t(path);
+    else
+#endif
+        ifile = new std::ifstream(path, std::ifstream::binary);
```

Comment:
This statement hanging off the else in the #ifdef sounds like trouble. Maybe pull the declaration of is_gzipped out of the first ifdef and check it here again?

Reply:
SGTM.  Done.